src/robot/libraries/BuiltIn.py

```diff
@@ -613,9 +613,16 @@ def should_be_equal(self, first, second, msg=None, values=True):
         for example, string ``false`` or ``no values``. See `Boolean arguments`
         section for more details.
 
+        ``ignore_case`` is False by default.  It is a boolean value, and if True
+        indicates that 'first' and 'second' should be compared case-insensitively,
+        provided that 'first' and 'second' are string types.
+
         If both arguments are multiline strings, the comparison is done using
         `multiline string comparisons`.
         """
+        if ignore_case and is_string(first) and is_string(second):
+            first = str(first).lower()
+            second = str(second).lower()
```

Comment:
Must not use `str()` here because it won't handle non-ASCII characters in Python 2. If we know values are strings, we can simply do `first = first.lower()`.  Adding a test for non-ASCII stuff  like I proposed in another comment is a good idea.